test/components/DateInput_spec.jsx

```diff
@@ -89,6 +89,28 @@ describe('DateInput', () => {
     });
   });
 
+  describe('#componentWillReceiveProps', () => {
+    describe('nextProps.displayValue exists', () => {
+      it('sets state.displayString to \'\'', () => {
+        const dateString = 'foo123';
+        const wrapper = shallow(<DateInput id="date" />).dive();
+        wrapper.setState({ dateString });
+        wrapper.instance().componentWillReceiveProps({ displayValue: '1991-07-13' });
+        expect(wrapper.state().dateString).to.equal('');
```

Comment:
`expect(wrapper.state()).to.have.property('dateString', '')` has a better failure message